packages/tocco-ui/src/Upload/Upload.js

```diff
@@ -9,47 +9,43 @@ import UploadInput from './UploadInput'
 /**
  * Component to upload files. If uploaded, the file will be displayed by the Preview component.
  */
-export class Upload extends React.Component {
-  constructor(props) {
-    super(props)
-    this.state = {isUploading: false}
-  }
+export const Upload = props => {
+  const {
+    value,
+    textResources,
+    onUpload
+  } = props
+  const [preview, setPreview] = useState({isUploading: false})
 
-  setUploadingState(file) {
-    this.setState({isUploading: true, previewFile: file})
+  const setUploadingState = file => {
+    setPreview({isUploading: true, previewFile: file})
   }
 
-  abortUploadingState() {
-    this.setState({isUploading: false, previewFile: null})
+  const abortUploadingState = () => {
+    setPreview({isUploading: false, previewFile: null})
   }
 
-  onDrop(file) {
-    this.setUploadingState(file)
-    this.props.onUpload(file)
+  const onDrop = file => {
+    setUploadingState(file)
+    onUpload(file)
   }
 
-  componentDidUpdate(prevProps) {
-    if (!_isEqual(prevProps.value, this.props.value)) {
-      this.abortUploadingState()
+  useEffect(() => {
+    if (!_isEqual(value)) {
+      abortUploadingState()
     }
-  }
+  }, [value])
 
-  getContent(props) {
-    if (props.value && props.value.binaryLink) {
-      return <View {...props} deleteTitle={props.textResources.delete} downloadTitle={props.textResources.download} />
-    } else {
-      if (this.state.isUploading) {
-        return <UploadProgress file={this.state.previewFile} text={props.textResources.uploading} {...props}/>
-      } else {
-        return <UploadInput {...props} text={props.textResources.upload} onDrop={this.onDrop.bind(this)}/>
-      }
+  const getContent = props => {
```

Comment:
imo it doesn't make sense to pass props here. props is in the closure of getContent? Especially makes no sense since you change props.value to value. So you are not using the parameter anyways.